cmd/skopeo/utils.go

```diff
@@ -1,26 +1,26 @@
 package main
 
 import (
-	"errors"
 	"fmt"
 	"strings"
 
-	"github.com/containers/image/directory"
-	"github.com/containers/image/docker"
-	"github.com/containers/image/image"
-	"github.com/containers/image/openshift"
+	topImage "github.com/containers/image"
 	"github.com/containers/image/types"
 	"github.com/urfave/cli"
 )
 
-const (
-	// atomicPrefix is the URL-like schema prefix used for Atomic registry image references.
-	atomicPrefix = "atomic:"
-	// dockerPrefix is the URL-like schema prefix used for Docker image references.
-	dockerPrefix = "docker://"
-	// directoryPrefix is the URL-like schema prefix used for local directories (for debugging)
-	directoryPrefix = "dir:"
-)
+// parseImageName converts URL-like image name to an types.ImageReference.
+func parseImageName(imgName string) (types.ImageReference, error) {
```

Comment:
Done, creating `containers/image/transports.ParseImageName`. The `transports` package is, sadly, probably necessary.